lib/explorer/backend/lazy_series.ex

```diff
@@ -350,6 +351,31 @@ defmodule Explorer.Backend.LazySeries do
     Backend.Series.new(data, dtype)
   end
 
+  @impl true
+  def select(%Series{} = predicate, %Series{} = on_true, %Series{} = on_false) do
+    args = [
+      series_or_lazy_series!(predicate),
+      series_or_lazy_series!(on_true),
+      series_or_lazy_series!(on_false)
+    ]
+
+    data = new(:select, args, aggregations?(args), window_functions?(args))
+
+    dtype =
+      cond do
+        on_true.dtype in [:float, :integer] ->
+          resolve_numeric_dtype([on_true, on_false])
+
+        on_true.dtype != on_false.dtype ->
+          raise "on_true dtype #{on_true.dtype} must equal on_false dtype #{on_false.dtype}"
```

Comment:
We don't need to check this here because Series.select already checks they both match or not. So we only need the first and third clause (it can be a regular if/else). :)

Reply:
Ah ya makes sense. tbh i'm new to Elixir and to Rust, haven't fully grokked the codebase. I was mostly just pattern matching my way through this PR. Totally makes sense now that you mention it.